### ISSUE TITLE
Maximum number of items in history does not seem to be removing items

### ISSUE THREAD

**Issue by robertstrom** [hluk/CopyQ#1749]
Before creating a new issue, see list of [known
issues](https://copyq.readthedocs.io/en/latest/known-issues.html).

**Describe the bug**
Copyq not remove clip items after the maximum number of history items has been reached / exceeded. My maximum number of history items is set to 2500 yet I have 9999 items in my copyq clipboard.

**To Reproduce**
1. Go to Preferences / History / Maximum number of items in history / set to 2500
2. Click on OK

**Expected behavior**
Clear out oldest clip history items once maximum number of items has been reached.

**Screenshots**
![Preferences — CopyQ_001](https://user-images.githubusercontent.com/3240205/132169112-31e6e1a2-676f-4163-b522-45542e2fae3d.png)
![About — CopyQ_001](https://user-images.githubusercontent.com/3240205/132169115-d2005cd7-def5-48e0-b328-377badaff168.png)
![Distributor ID: Pop   (5 lines) — CopyQ_001](https://user-images.githubusercontent.com/3240205/132169419-23b7da0a-e071-44a7-a6cc-9ba195b1a3ac.png)

**Version, OS and Environment**
(Get details from `copyq version` command if possible.)
 - Application Version 4.1.0
 - OS - Linux Pop!_OS (based on Ubuntu)
      Distributor ID:	Pop
      Description:	Pop!_OS 21.04
      Release:	21.04
      Codename:	hirsute

 - Desktop environment, window manager (if applicable)
GNOME

**Comment by hluk** [comment]
Can you check if "Maximum number of items" is not overridden for the tab (it's under "Tab" in Preferences)?

**Comment by robertstrom** [comment]
Good call ... you are absolutely right. I did have that set to ... take a guess ... 9999

Sorry for the false alarm. I will reduce the number and see how it behaves.

Thanks,
Robert

**Comment by robertstrom** [comment]
I changed the number to 9995 and it changed the number stored immediately.

I changed the number to 5000 and it changed the number immediately.

Command line reports the same number

`rstrom@rstrom-XPS-15-9550:~$ copyq count`
`5000`
`rstrom@rstrom-XPS-15-9550:~$ `

Thanks again!

Luv this program!! I do not know how long you have been working on Copyq, but I have been using a clipboard manger on Windows (Clipmate) since the mid to late 1990's and the is the only clipboard manager program that rivals it. There has been no development on Clipmate for quite a long time and it is only a Windows program. FYI - Copyq = so awesome I have even donated ;-)
